samsts/fill.py

```diff
@@ -30,24 +31,36 @@ def fill_scene(
     requires a cluster_pix_idxs dictionary (see cluster.assign_to_cluster) and
     a cluster_neighbors dictionary (see cluster.find_cluster_neighbors).
     """
+    cache_fn = cache.SCENE_FILL.format(scene=scene)
+    if cache.exists(cache_fn):
+        # TODO dtype
+        return cache.read(cache_fn)
+
     scene_idxs = stack.get_scene_idxs(scene)
+    scene_idx = scene_idxs[0]
     total_gaps = (
         stack.footprint_mask & np.isnan(stack.matrix[:, :, scene_idxs[0]])
```

Comment:
```suggestion
        stack.footprint_mask & np.isnan(stack.matrix[:, :, scene_idx])
```

if we got a var we should use it!